mathics/core/number.py

```diff
@@ -2,28 +2,33 @@
 # cython: language_level=3
 
 import string
-from math import ceil, log, log2
+from math import ceil, log
+from sys import float_info
 from typing import List, Optional
 
 import mpmath
 import sympy
 
+from mathics.core.element import BaseElement
 from mathics.core.symbols import (
     SymbolMachinePrecision,
     SymbolMaxPrecision,
     SymbolMinPrecision,
 )
 
-C = log2(10)  # ~ 3.3219280948873626
+C = mpmath.log(10.0, 2.0)  # ~ 3.3219280948873626
 
-# Number of bits of machine precision.
-# Note this is a float, not an int.
-# WMA uses real values for precision, to take into account the internal representation of numbers.
-# This is why $MachinePrecision is not 16, but 15.9546`
-machine_precision = 53.0
-machine_digits = int(machine_precision / C)
+machine_precision = float_info.mant_dig
+machine_digits = float_info.dig
 
-machine_epsilon = 2 ** (1 - machine_precision)
+MACHINE_EPSILON = float_info.epsilon
+MACHINE_PRECISION_VALUE = float_info.mant_dig / C
+MIN_MACHINE_NUMBER = float_info.min
+MAX_MACHINE_NUMBER = float_info.max
+ZERO_MACHINE_ACCURACY = -mpmath.log(MIN_MACHINE_NUMBER, 10.0) + MACHINE_PRECISION_VALUE
+
+# backward compatibility
```

Comment:
Is this necessary? This release we are bumping the major number. If so, suggest possibly in the comment when we can remove this. (I would find it hard to believe that anyone other than ourselves is making use of this.)

Reply:
I leave this commentary because that variable appears in several modules. I can make the change, but then this PR would become larger. Also, I wondered if it was worth making the change. Thoughts?

Reply:
I think it worth making the change. If you want to do in another PR, that's okay.

Reply:
Then, I will do that in the following PR.